crates/threshold-bls/src/sig/sig.rs

```diff
@@ -61,7 +58,16 @@ pub trait SignatureScheme: Scheme {
     type Error: Error;
 
     fn sign(private: &Self::Private, msg: &[u8]) -> Result<Vec<u8>, Self::Error>;
+
+    fn sign_without_hashing(private: &Self::Private, msg: &[u8]) -> Result<Vec<u8>, Self::Error>;
```

Comment:
What's the point at all of having this ? 
I'm really worried now about the a-priori un-necessary public methods you are exposing in this PR. I paid special attention that all exposed method were uniquely the ones a client needed to have. 
What's the goal of this PR ?

Reply:
In particular, why would a user need to sign something without hashing, what's the use case you are targeting ?

Reply:
In particular, why would a user need to sign something without hashing, what's the use case?

Reply:
My understanding is that when blind signing you'd expect an already "hashed" (hash+ blinding factor) message, while when doing normal signing you'd hash internally. As a result, our services which will be signing blinded messages should _not_ hash, while a user who's signing something for themselves _should_ hash

cc @kobigurk

Reply:
@gakonst is right - when you partially sign a blinded message, you can't hash since you don't have access to the message - you already receive a blinded hashed point and need to just sign that, to keep the linear properties that allow it to be unblinded later on.

Reply:
Yes but that behavior was there before without needed to expose additional methods:
* [Here](https://github.com/celo-org/celo-threshold-bls-rs/blob/00b69c715f9a00d947fb0efae923d0cd5bfb0f59/examples/tblind/node.rs#L107) a node receives a blinded message and signs it with `sign_partial`
* [Here](https://github.com/celo-org/celo-threshold-bls-rs/blob/00b69c715f9a00d947fb0efae923d0cd5bfb0f59/src/sig/blind.rs#L167) the call `sign()` expects a blinded message already.

I mean that the behavior here was hidden in the API, but these changes are exposing them which is worried imo.

Reply:
- `sign` could do that because the Blinder was [signing without hashing](https://github.com/celo-org/celo-threshold-bls-rs/blob/00b69c715f9a00d947fb0efae923d0cd5bfb0f59/src/sig/blind.rs#L52-L57)

Reply:
I made an "emergency patch" to be able to move forward to remove the hashing

Reply:
This is the relevant patch: https://github.com/celo-org/celo-threshold-bls-rs/commit/921f374777a6e4d0fbbe2736845fcfb241d6c94b#diff-38e36722cadd105c46f1a4cd95de6f07L19-R35